src/Section.cpp

```diff
@@ -47,7 +47,7 @@ Section::Section(shared_ptr<base::ISection> &&ptr)
 
 
 void Section::link(const Section &link) {
-    if (link == none) {
+    if (!link) {
         backend()->link(none);
     } else {
         backend()->link(link.id());
```

Comment:
no match for ternary ‘operator?:’ (operand types are ‘bool’, ‘std::string {aka std::basic_string<char>}’, and ‘const none_t {aka int boost::detail::none_helper::* const}’)